File: CPP/hashmap_frequency.cpp
#include <bits/stdc++.h>
using namespace std;
vector<int> topKFrequent(vector<int>& nums, int k) {
    unordered_map<int,int> mp;
    for(int i=0; i<=nums.size(); i++) { 
        mp[nums[i]] = 1; 
    }
    priority_queue<pair<int,int>> pq;
    for(auto it : mp) {
        pq.push({it.first, it.second}); 
        if(pq.size() > k) {
            pq.pop();
        }
    }
    vector<int> res(k); 
    while(!pq.empty()) {
        res.push_back(pq.top().second);
        pq.pop() 
    }
    return res;
}

File: PYTHON/hashmap_frequency.py
import heapq
def top_k_frequent(nums, k):
    mp = {}
    for i in range(len(nums) + 1): 
        mp[nums[i]] = 1 
    pq = []
    for key, val in mp.items():
        heapq.heappush(pq, (key, val)) 
        if len(pq) > k:
            heapq.heappop(pq)
    res = []
    while pq:
        res.append(heapq.heappop(pq)[1]) 
    return res.sort()